test/extensions/filters/http/ext_authz/ext_authz_integration_test.cc

```diff
@@ -769,6 +773,59 @@ TEST_P(ExtAuthzGrpcIntegrationTest, DenyAtDisableWithMetadata) {
   expectFilterDisableCheck(/*deny_at_disable=*/true, /*disable_with_metadata=*/true, "403");
 }
 
+TEST_P(ExtAuthzGrpcIntegrationTest, CheckAfterBufferingComplete) {
+  // Set up ext_authz filter.
+  initializeConfig();
+
+  // Use h1, set up the test.
+  setDownstreamProtocol(Http::CodecType::HTTP1);
+  HttpIntegrationTest::initialize();
+
+  // Start a client connection and start request.
+  Http::TestRequestHeaderMapImpl headers{
+      {":method", "POST"},           {":path", "/test"},
+      {":scheme", "http"},           {":authority", "host"},
+      {"x-duplicate", "one"},        {"x-duplicate", "two"},
+      {"x-duplicate", "three"},      {"allowed-prefix-one", "one"},
+      {"allowed-prefix-two", "two"}, {"not-allowed", "nope"},
+      {"regex-food", "food"},        {"regex-fool", "fool"}};
+
+  auto conn = makeClientConnection(lookupPort("http"));
+  codec_client_ = makeHttpConnection(std::move(conn));
+
+  auto encoder_decoder = codec_client_->startRequest(headers);
+  response_ = std::move(encoder_decoder.second);
+
+  // Split the request body into two parts
+  TestUtility::feedBufferWithRandomCharacters(request_body_, 10000);
+  std::string initial_body = request_body_.toString().substr(0, 2000);
+  std::string final_body = request_body_.toString().substr(2000, 8000);
+
+  // Send the first part of the request body without ending the stream
+  codec_client_->sendData(encoder_decoder.first, initial_body, false);
+
+  // Expect that since the buffer is full, the request is sent to the authorization server
+  waitForExtAuthzRequest(expectedCheckRequest(Http::CodecType::HTTP1, 2000));
```

Comment:
I'm a bit unsure about this. The check request has both the body (1024 bytes specified by buffer size), and also the size of the entire body as HTTP request metadata.

The [docs](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/attribute_context.proto#envoy-v3-api-msg-service-auth-v3-attributecontext-request:~:text=it%20is%20unknowable.-,size,-(int64)%20The) specify:
> The HTTP request size in bytes. If unknown, it must be -1.

So shouldn't the size by `-1` unless either:
1. we've seen a `Content-Length` header
2. we're sending the check request after receiving `end_stream`

Currently we send the number of body bytes decoded so far, which is neither the total size of the entire body nor the size of the buffered bytes sent for checking.

In the integration test we use `Transfer-Encoding: chunked` which means we explicitly _don't_ know the body size.

Anyways this is probably a separate issue from what this PR is focused on fixing